integrations/tensorflow/e2e/iree_e2e_product_test_suite.bzl

```diff
@@ -0,0 +1,229 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Macro for building e2e tests from a single source with multiple flags."""
+
+load("//bindings/python:build_defs.oss.bzl", "iree_py_test")
+
+def normalize_dictionary(dictionary):
+    """Wraps every value of dictionary in a list if it isn't one already."""
+    for key, value in dictionary.items():
+        if type(value) != type([]):
+            dictionary[key] = [value]
+    return dictionary
+
+def dictionary_product(dictionary):
```

Comment:
It seems like this going to do a lot of list copies, but I guess the inputs should be small, so maybe not a big deal. You know ahead of time how many dictionaries there are going to be though, so it seems like it should be possible to create them all ahead of time and then go through in a nested for loop.

Reply:
The runtime for `bazel query integrations/...` is less than 0.2s, and I imagine most of that is `bazel` overhead. I vote to leave it as-is for now since it's implemented quite simply and optimize things if the runtime becomes an issue.